package/src/constructs/construct.ts

```diff
@@ -1,20 +1,22 @@
 import { Session } from './project'
 import { Ref } from './ref'
 
-export class Construct {
+export abstract class Construct {
+  type: string
   logicalId: string
-  constructor (logicalId: string) {
+  constructor (type: string, logicalId: string) {
+    Session.validateCreateConstruct(this)
```

Comment:
Why validate this here, rather than in `Session.registerConstruct`?

If a user is trying to create an unsupported construct (f.ex. a `BrowserCheck`) in the checkly config file, some necessary `Session` properties won't be configured. Validating that we can register the construct early on makes sure that the user get's a clear error, rather than an error about a missing `Session` property - f.ex:
 https://github.com/checkly/checkly-cli/blob/827cf71b4e8a46648c0763a2ad685166948439b9/package/src/constructs/browser-check.ts#L59